server/src/main/java/io/crate/expression/reference/doc/lucene/LuceneReferenceResolver.java

```diff
@@ -218,6 +187,19 @@ public LiteralValueExpression(Object value) {
         public Object value() {
             return value;
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
```

Comment:
Why did you add equals and hashCode, is it needed now?

Reply:
It made writing the test easier, but I can remove it and rework the test if you think that's better?

Reply:
I just didn't see equals and hashcode on other classes, so If it's only for the tests I'd say either change the test or add a comment here to explain where they are used.